### PR TITLE
chore: complete editor config for Markdown files

### PR DESCRIPTION
**Pull request by themr0c**
* Added more complete editor config for Markdown files

NB: I haven't found a way to configure for one sentence per line aka. ventilated prose.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -15,4 +15,22 @@ indent_size = 2
 indent_style = space
 
 [*.md]
-trim_trailing_whitespace = false
+max_line_length = 999
+tab_width = 2
+ij_continuation_indent_size = 2
```
Comment (benoitf) [inline .editorconfig:20]: it looks like specific annotations for intellij ? so not sure if it should be there. It should contains only generic properties.

the formatter used is prettier so if it requires some formatting check/format it should probably goes there

Comment (themr0c) [reply .editorconfig:20]: It's a way to achieve consistency. Some features work only on IntelliJ. I am happy to provide similar features for VSCode and other.

Comment (themr0c) [reply .editorconfig:20]: Prettier is something I would like to explore more, but I am not sure it integrates as well with the IDE.

Comment (themr0c) [reply .editorconfig:20]: NB: the Prettier plugin is not available on IntelliJ IDEA. https://youtrack.jetbrains.com/issue/WEB-59644/Provide-Prettier-plugin-on-IntelliJ-IDEA That's such bad news.

Comment (themr0c) [reply .editorconfig:20]: I had a look at Prettier documentation: I have no doubts it is a splendid tool for the code (it enables me to dare to do some contributions), but we can't rely on it to lint/format Markdown in documentation. 

* It supports EditorConfig for basic stuff https://prettier.io/docs/en/configuration.html#editorconfig

* The only prose related option is better kept as default (preserve): https://prettier.io/docs/en/options.html#prose-wrap

We have yet to find a good way to reformat Markdown files without breaking nested blocks.

**Review by benoitf — CHANGES_REQUESTED** [review]
we can't add properties that will work only if you're using intellij

Either we have generic properties or we use a formatter with a check/fix command that is independent of the IDE being used.

**Comment by benoitf** [comment]
closing as no follow-up

**Pull request closed without merging**